modules/repository-azure/src/test/java/org/elasticsearch/repositories/azure/AzureStorageServiceTests.java

```diff
@@ -186,25 +186,18 @@ public void testReinitClientWrongSettings() throws IOException {
         secureSettings1.setString("azure.client.azure1.account", "myaccount1");
         secureSettings1.setString("azure.client.azure1.key", encodeKey("mykey11"));
         final Settings settings1 = Settings.builder().setSecureSettings(secureSettings1).build();
+
         final MockSecureSettings secureSettings2 = new MockSecureSettings();
-        secureSettings2.setString("azure.client.azure1.account", "myaccount1");
-        // missing key
+        secureSettings2.setString("azure.client.azure1.account", "myaccount3");
+        secureSettings2.setString("azure.client.azure1.key", encodeKey("mykey33"));
+        secureSettings2.setString("azure.client.azure1.sas_token", encodeKey("mysasToken33"));
         final Settings settings2 = Settings.builder().setSecureSettings(secureSettings2).build();
-        final MockSecureSettings secureSettings3 = new MockSecureSettings();
-        secureSettings3.setString("azure.client.azure1.account", "myaccount3");
-        secureSettings3.setString("azure.client.azure1.key", encodeKey("mykey33"));
-        secureSettings3.setString("azure.client.azure1.sas_token", encodeKey("mysasToken33"));
-        final Settings settings3 = Settings.builder().setSecureSettings(secureSettings3).build();
         try (AzureRepositoryPlugin plugin = pluginWithSettingsValidation(settings1)) {
             final AzureStorageService azureStorageService = plugin.azureStoreService.get();
             AzureBlobServiceClient client11 = azureStorageService.client("azure1", LocationMode.PRIMARY_ONLY);
             assertThat(client11.getSyncClient().getAccountUrl(), equalTo("https://myaccount1.blob.core.windows.net"));
             assertThat(
                 expectThrows(SettingsException.class, () -> plugin.reload(settings2)).getMessage(),
-                is("Neither a secret key nor a shared access token was set for account [myaccount1]")
```

Comment:
What happens if there is no access, i.e., also no managed identity setup? I wonder if we need a test for that to demonstrate that it fails in a good enough way?

Reply:
The error is actually pretty descriptive, much better than what we get with the AWS SDK, but yeah you're right we should have a test for that too.

Reply:
See 
f705e34